### PR TITLE
Add ecoCode-csharp plugin

### PR DESCRIPTION
**Pull request by dedece35**
Hello!
This PR adds CSharp plugin of ecoCode project into SonarQube Marketplace.

Project key: ecocodecsharp
Version: 1.1.0 (supports SonarQube 9.9.5 LTS to 10.5.1)
Project URL: https://github.com/green-code-initiative/ecoCode-csharp-sonarqube
Test project: https://github.com/green-code-initiative/ecoCode-csharp-test-project
SonarCloud: https://sonarcloud.io/project/overview?id=green-code-initiative_ecoCode-csharp-sonarqube

Thank you!
Best regards

### CODE REVIEW

**Comment by dedece35** [comment]
Hello @ganncamp, do you have some time to make a review ? ...
We have our yearly "ecoCode" challenge at Paris during 2 days with 100 developers, tomorrow and the day after.
We would be proud to announce that the new plugin is on de official Sonarqube marketPlace :p 

thank you a lot !

**Comment by ganncamp** [comment]
I was off last week. This is on my list.

**Comment by ganncamp** [comment]
For an update of an existing plugin, no thread is required, but for a brand new plugin, I need a thread in the community, please. https://community.sonarsource.com/

**Comment by dedece35** [comment]
> https://github.com/green-code-initiative/ecoCode-csharp-test-project

hello @ganncamp 
here is the new topic https://community.sonarsource.com/t/new-plugin-ecocode-csharp-requesting-inclusion-in-sonarqube-marketplace/116726

thank you.

**Review by ganncamp — CHANGES_REQUESTED** [review]
For marketplace entry, you also need to register the plugin in the list of plugins: https://github.com/SonarSource/sonar-update-center-properties/blob/e7ee53ea68b65f39b7e2a10ff03260d75ada1729/update-center-source.properties#L1

**Comment by dedece35** [comment]
> For marketplace entry, you also need to register the plugin in the list of plugins:
> 
> https://github.com/SonarSource/sonar-update-center-properties/blob/e7ee53ea68b65f39b7e2a10ff03260d75ada1729/update-center-source.properties#L1

Hi @ganncamp,

modification done and ready in this PR.
thank you.

File: update-center-source.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-plugins=aemrules,ansible,apigee,authaad,authgoogleoauth,authoidc,checkstyle,clover,codehawk,communitybsl,communitydelphi,communitygosu,communityrust,dependencycheck,devon4j,ecocodeandroid,ecocodejava,ecocodejavascript,ecocodephp,ecocodepython,findbugs,groovy,hadolint,ibmixaemrules,jdepend,jqassistant,l10nde,l10nfr,l10nel,l10nes,l10nit,l10nja,l10nko,l10nru,l10nzh,l10nzhtw,mutationanalysis,mybatis,openedge,pmd,shellcheck,rci,scmcvs,scmjazzrtc,scmtfvc,scmmercurial,softvis3d,sonargraphintegration,webdriver,yaml
```
Comment (ganncamp) [inline update-center-source.properties:1]: I hate to be picky, but could you put this in alpha order, please?

**Review by ganncamp — CHANGES_REQUESTED** [review]

File: update-center-source.properties (hunk 1 above)
Comment (dedece35) [reply update-center-source.properties:1]: you are right ! I'm picky like you :p with my reviews.
update done.

**Review by ganncamp — APPROVED** [review]

**Pull request merged**